modules/dashboard/cmd/main.go

```diff
@@ -145,15 +177,17 @@ func main() {
 			defer cancel()
 
 			var wg sync.WaitGroup
-			wg.Add(2)
+			wg.Add(len(servers) + 1) // +1 for app shutdown
```

Comment:
How about calling `wg.Add()` before each goroutine is registered so we don't have to pre-determine the number here?